src/shared/components/post/post-listing.tsx

```diff
@@ -642,341 +508,469 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
+            {showScores() && (
+              <span class="ml-2">{numToSI(this.state.upvotes)}</span>
+            )}
+          </button>
+          {this.props.enableDownvotes && (
+            <button
+              className={`ml-2 btn-animate btn py-0 px-1 ${
+                this.state.my_vote == -1 ? "text-danger" : "text-muted"
+              }`}
+              onClick={linkEvent(this, this.handlePostDisLike)}
+              {...tippy}
+              aria-label={i18n.t("downvote")}
+            >
+              <Icon icon="arrow-down1" classes="icon-inline small" />
+              {showScores() && (
+                <span
+                  class={classNames("ml-2", {
+                    invisible: this.state.downvotes === 0,
+                  })}
                 >
-                  <Icon
-                    icon="star"
-                    classes={`icon-inline ${post_view.saved && "text-warning"}`}
-                  />
-                </button>
+                  {numToSI(this.state.downvotes)}
+                </span>
               )}
-              <Link
-                className="btn btn-link btn-animate text-muted py-0"
-                to={`/create_post${this.crossPostParams}`}
-                title={i18n.t("cross_post")}
-              >
-                <Icon icon="copy" classes="icon-inline" />
-              </Link>
-              {!this.myPost && (
-                <>
+            </button>
+          )}
+        </div>
+      </>
+    );
+  }
+
+  get saveButton() {
+    let saved = this.props.post_view.saved;
+    let label = saved ? i18n.t("unsave") : i18n.t("save");
+    return (
+      <button
+        class="btn btn-link btn-animate text-muted py-0"
+        onClick={linkEvent(this, this.handleSavePostClick)}
+        data-tippy-content={label}
+        aria-label={label}
+      >
+        <Icon
+          icon="star"
+          classes={classNames({ "text-warning": saved })}
+          inline
+        />
+      </button>
+    );
+  }
+
+  get crossPostButton() {
+    return (
+      <Link
+        className="btn btn-link btn-animate text-muted py-0"
+        to={`/create_post${this.crossPostParams}`}
+        title={i18n.t("cross_post")}
+      >
+        <Icon icon="copy" inline />
+      </Link>
+    );
+  }
+
+  get reportButton() {
+    return (
+      <button
+        class="btn btn-link btn-animate text-muted py-0"
+        onClick={linkEvent(this, this.handleShowReportDialog)}
+        data-tippy-content={i18n.t("show_report_dialog")}
+        aria-label={i18n.t("show_report_dialog")}
+      >
+        <Icon icon="flag" inline />
+      </button>
+    );
+  }
+
+  get blockButton() {
+    return (
+      <button
+        class="btn btn-link btn-animate text-muted py-0"
+        onClick={linkEvent(this, this.handleBlockUserClick)}
+        data-tippy-content={i18n.t("block_user")}
+        aria-label={i18n.t("block_user")}
+      >
+        <Icon icon="slash" inline />
+      </button>
+    );
+  }
```

Comment:
I have some preferences for which ones of these get shown by default on the full view, the mobile view, and which ones are hidden behind the 3 dot menu, but again we can wait for #449